src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -377,8 +379,8 @@ export class SavedObjectsService
      * HTTP server running without an Elasticsearch server being available.
      * So, when the `migrations.skip` is true, we skip migrations altogether.
      */
-    const cliArgs = this.coreContext.env.cliArgs;
-    const skipMigrations = cliArgs.optimize || this.config.migration.skip;
+    const skipMigrations =
+      this.config.migration.skip || this.coreContext.env.isDevClusterMaster || !pluginsInitialized;
```

Comment:
`cliArgs.optimize` was the wrong option to be listening for. This CLI arg is only specified when running kibana to _only_ run the optimizer and then exit. It is not set when running the optimizer in dev. However, in both cases, plugin initialization is disabled which is really the case where we shouldn't run migrations.